vkconfig_core/util.h

```diff
@@ -55,7 +55,7 @@ bool IsNumber(const std::string& s);
 
 template <typename T>
 typename std::vector<T>::iterator FindItByKey(std::vector<T>& container, const char* key) {
-    assert(key != nullptr && key != "");
+    assert(key != nullptr);
```

Comment:
Is the change in behavior intentional?  The original assertion could be rewritten as something like:
```
assert(key != nullptr && key[0] != '\0');
```
or:
```
assert(key != nullptr && std::strcmp(key, "") != 0);
```
right?

Reply:
Yes expect, key != "" cause a warning as this is undefined

Reply:
I do see why `key != ""` would not work.  But removing the check entirely changes behavior, which I don't expect from a change intended only to fix compiler errors, as there were other ways to change the code that would have preserved the original behavior.

If the behavior change is intentional, I'm okay with it.

Reply:
Why was the check for an empty string removed entirely, instead of being fixed to use `std::strcmp`?

Reply:
Ah good point, my bad. I've fixed it with upcoming PR.